test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_kernel.cpp

```diff
@@ -0,0 +1,1453 @@
+
+        cl_int error;
+        clProgramWrapper program;
+        clKernelWrapper kernel;
+        size_t threads[1], localThreads[1];
+        std::vector<cl_int> constantData;
+        std::vector<cl_int> resultData;
+
+        error =
+            create_single_kernel_helper(context, &program, &kernel, 1,
+                                        sample_const_arg_kernel, "sample_test");
+        test_error(error, "Creating kernel failed");
+
+        currentSize = max_size;
+        MTdataHolder d(gRandomSeed);
+
+        size_t sizeToAllocate =
+            ((size_t)currentSize / sizeof(cl_int)) * sizeof(cl_int);
+        size_t numberOfInts = sizeToAllocate / sizeof(cl_int);
+        constantData.resize(sizeToAllocate / sizeof(cl_int));
+        resultData.resize(sizeToAllocate / sizeof(cl_int));
+
+        for (int i = 0; i < (int)(numberOfInts); i++)
+            constantData[i] = (int)genrand_int32(d);
+
+        clMemWrapper streams[3];
+        streams[0] =
+            clCreateBuffer(context, CL_MEM_COPY_HOST_PTR, sizeToAllocate,
+                           constantData.data(), &error);
+        test_error(error, "Creating test array failed");
+        streams[1] = clCreateBuffer(context, CL_MEM_READ_WRITE, sizeToAllocate,
+                                    NULL, &error);
+        test_error(error, "Creating test array failed");
+
+
+        /* Set the arguments */
+        error = clSetKernelArg(kernel, 0, sizeof(streams[0]), &streams[0]);
+        test_error(error, "Unable to set indexed kernel arguments");
+        int intarg = 10;
+        error = clSetKernelArg(kernel, 1, sizeof(int), &intarg);
+        test_error(error, "Unable to set indexed kernel arguments");
+
+        threads[0] = numberOfInts;
+        localThreads[0] = 1;
+
+        cl_ndrange_kernel_command_properties_khr props[] = {
+            CL_MUTABLE_DISPATCH_UPDATABLE_FIELDS_KHR,
+            CL_MUTABLE_DISPATCH_ARGUMENTS_KHR, 0
+        };
+
+        error = clCommandNDRangeKernelKHR(
+            mutable_command_buffer, nullptr, props, kernel, 1, nullptr, threads,
+            localThreads, 0, nullptr, nullptr, &command);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(mutable_command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, mutable_command_buffer, 0,
+                                          nullptr, nullptr);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        streams[1] = clCreateBuffer(context, CL_MEM_READ_WRITE, sizeToAllocate,
+                                    NULL, &error);
+        test_error(error, "Creating test array failed");
+
+        clMemWrapper newBuffer;
+        newBuffer = clCreateBuffer(context, CL_MEM_READ_WRITE, sizeToAllocate,
+                                   NULL, &error);
+
+        cl_mutable_dispatch_arg_khr arg_1{ 1, sizeof(streams[1]) * 2, nullptr };
```

Comment:
Should be `sizeof(int)` I think and pass a new int value, such as `intarg`  earlier.